x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/store/NativePrivilegeStoreTests.java

```diff
@@ -114,42 +118,27 @@ public void cleanup() {
     }
 
     public void testGetSinglePrivilegeByName() throws Exception {
-        final ApplicationPrivilegeDescriptor sourcePrivilege = new ApplicationPrivilegeDescriptor("myapp", "admin",
-            newHashSet("action:admin/*", "action:login", "data:read/*"), emptyMap()
-        );
-
-        final PlainActionFuture<ApplicationPrivilegeDescriptor> future = new PlainActionFuture<>();
-        store.getPrivilege("myapp", "admin", future);
-        assertThat(requests, iterableWithSize(1));
-        assertThat(requests.get(0), instanceOf(GetRequest.class));
-        GetRequest request = (GetRequest) requests.get(0);
-        assertThat(request.index(), equalTo(RestrictedIndicesNames.SECURITY_MAIN_ALIAS));
-        assertThat(request.id(), equalTo("application-privilege_myapp:admin"));
-
-        final String docSource = Strings.toString(sourcePrivilege);
-        listener.get().onResponse(new GetResponse(
-            new GetResult(request.index(), request.id(), 0, 1, 1L, true,
-                new BytesArray(docSource), emptyMap(), emptyMap())
+        final List<ApplicationPrivilegeDescriptor> sourcePrivileges = List.of(
+            new ApplicationPrivilegeDescriptor("myapp", "admin",
+                newHashSet("action:admin/*", "action:login", "data:read/*"), emptyMap()
         ));
-        final ApplicationPrivilegeDescriptor getPrivilege = future.get(1, TimeUnit.SECONDS);
-        assertThat(getPrivilege, equalTo(sourcePrivilege));
-    }
 
-    public void testGetMissingPrivilege() throws Exception {
```

Comment:
Why did this test get removed?
Isn't that still a valid scenario?

Reply:
The method under tests, `NativePrivilegeStore#getPrivilege`, is removed. So there test is no longer necessary.

`NativePrivilegeStore#getPrivilege` is removed because the code now always fetch all privileges for an application for caching purpose.

Reply:
Do we have a test somewhere for 
```
public void getPrivileges(Collection<String> applications, 
                          Collection<String> names,
                          ActionListener<Collection<ApplicationPrivilegeDescriptor>> listener)
```

when the application exists but none of the privilege names do? That feels like something worth testing.

Reply:
I added back a modified version of `testGetMissingPrivilege`. But then I think it is not really necessary. Please let me explain:

The code now always issue a search request, from end result's perspective, the difference really is just how many SearchHit is returned. It could be 0, 1 or N. What you described here is a case of `0`. There are other cases of `0` hits, e.g. non-existent application, but they are not different in terms of search response.

Search response is turned into a collection of `ApplicationPrivilegeDescriptor` by `buildPrivilege` and `ScrollHelper#fetchAllByEntity`. What we do want test is `buildPrivilege` since it is part of the class and it is only relevant when SearchHit is greater than `0`, because when there is no SearchHit, it won't get executed at all and we probably can trust `ScrollHelper` to do its job by returning an empty list.

I guess this is probably why both [`testGetPrivilegesByWildcardApplicationName`](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/store/NativePrivilegeStoreTests.java#L182) and [`testGetPrivilegesByStarApplicationName`](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/store/NativePrivilegeStoreTests.java#L202) did not bother to test the `future` results (since they are just empty lists). And one could say they already suffice to be tests for `0` SearchHit.

With above being said, I decided to keep `testGetMissingPrivilege` since it is already added and an extra test is not a bad thing.